controllers/platformoperator_controller.go

```diff
@@ -62,47 +67,82 @@ func (r *PlatformOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	log.Info("reconciling request", "req", req.NamespacedName)
 	defer log.Info("finished reconciling request", "req", req.NamespacedName)
 
-	// TODO: flesh out status condition management
 	po := &platformv1alpha1.PlatformOperator{}
 	if err := r.Get(ctx, req.NamespacedName, po); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 	defer func() {
 		po := po.DeepCopy()
 		po.ObjectMeta.ManagedFields = nil
+
 		if err := r.Status().Patch(ctx, po, client.Apply, client.FieldOwner("platformoperator")); err != nil {
 			log.Error(err, "failed to patch status")
 		}
 	}()
 
-	desiredBundle, err := r.Sourcer.Source(ctx, po)
+	return r.reconcile(ctx, po)
+}
+
+func (r *PlatformOperatorReconciler) reconcile(ctx context.Context, po *platformv1alpha1.PlatformOperator) (ctrl.Result, error) {
```

Comment:
What is the idea behind creating a second private version of reconcile? Could we capture that in a comment?

Reply:
This was an attempt at modularizing the reconciliation logic so it was easier to mock for unit testing purposes. Ended up deflating this logic back to the Reconcile method, and used e2e tests instead of bootstrapping a unit suite in #41.